### PR TITLE
Added wildcard support to paths

### PR DESCRIPTION
**Pull request by bjornandre**
The following PR adds the ability to browse the namespace paths that can be read via the dapla-spark-plugin. Example:

```scala
val datasett = spark.read.format("gsim").load("/skatt/person/*")
z.show(datasett)
```
This could produce the following output
```
+-------------------+
|               path|
+-------------------+
|/skatt/person/test1|
|/skatt/person/test2|
+-------------------+
```

### CODE REVIEW

**Comment by hadrienk** [comment]
Wildcard are supported in spark to create a dataframe that is a union of the matched files. This might confuse users.

**Comment by hadrienk** [comment]
I think we should look at https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```

**Comment by bjornandre** [comment]
@hadrienk I don´t see how the spark Catalog API will work, and it also seems like a big rewrite for little functionality.

I like the user-friendliness of using the same syntax for both browsing namespaces and loading data with minimal code change.

**Comment by hadrienk** [comment]
> @hadrienk I don´t see how the spark Catalog API will work, and it also seems like a big rewrite for little functionality.
> 
> I like the user-friendliness of using the same syntax for both browsing namespaces and loading data with minimal code change.

Agree, it does the job.

File: pom.xml [hunk 1]
```diff
@@ -151,6 +158,13 @@
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.spark</groupId>
+            <artifactId>spark-core_2.12</artifactId>
+            <version>2.4.5</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (hadrienk) [inline pom.xml]: Isn't this dependency transitively added by spark-sql?

File: src/main/java/no/ssb/dapla/spark/plugin/CatalogRelation.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    @Override
+    public StructType schema() {
+        return dataFrame.schema();
+    }
+
+    /**
+     * The equals should return true if it is known that the two relations will return the
+     * same data. Using the set of files guaranties this.
```
Comment (hadrienk) [inline src/main/java/no/ssb/dapla/spark/plugin/CatalogRelation.java]: The doc doesn't match the implementation.

File: src/main/java/no/ssb/dapla/spark/plugin/CatalogRelation.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        CatalogRelation that = (CatalogRelation) o;
+        return path.equals(that.path);
+    }
+
+    @Override
+    public int hashCode() {
+        return path.hashCode();
```
Comment (hadrienk) [inline src/main/java/no/ssb/dapla/spark/plugin/CatalogRelation.java]: Should be consistent with equals()

File: src/test/java/no/ssb/dapla/spark/plugin/CatalogRelationTest.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        // Read the unit dataset json example.
+        SparkSession session = SparkSession.builder()
+                .appName(GsimDatasourceLocalFSTest.class.getSimpleName())
+                .master("local")
+                .config("spark.ui.enabled", false)
+                .config(CONFIG_CATALOG_URL, baseUrl.toString())
+                .config(CONFIG_ROUTER_OAUTH_TOKEN_URL, "http://localhost") // not used
+                .config(CONFIG_ROUTER_OAUTH_CLIENT_ID, "na")
+                .config(CONFIG_ROUTER_OAUTH_CLIENT_SECRET, "na")
+                .config(CONFIG_ROUTER_OAUTH_TOKEN_IGNORE_EXPIRY, "true")
```
Comment (hadrienk) [inline src/test/java/no/ssb/dapla/spark/plugin/CatalogRelationTest.java]: This parameter is not in use anymore

File: pom.xml (hunk 1 above)
Comment (bjornandre) [reply pom.xml]: It's not transitively added when using spark-sql with provided scope.

**Pull request merged**